dvc/remote/base.py

```diff
@@ -165,7 +165,16 @@ def _collect_dir(self, path_info):
                 )
```

Comment:
If we are parallelizing it for the directory, why not just use ProcessPool here explicitly and submit self.get_file_checksum() to it? The current approach feels weird with that CHECKSUM_EXECUTOR thing.

Reply:
Also that would prevent you from isinstance(..., Future) checks too

Reply:
Aand it should also not require you to do any `Future`/`result()` adjustments to the test as well, and `file_md5` will stay straightforward and simple.

Reply:
When we implement it this was way it may parallelize implicitly, if we call `.result()` late enough for random spread everywhere `file_md5()` calls. I don't think it will work here for anything but `._collect_dir()`, so no point in complicating it.

Reply:
@Suor I don't quite get the point you are making :slightly_smiling_face: Could you explain, please?

Reply:
This PR only parallelizes `_collect_dir` at the moment, but there are many other callers of `file_md5`. With features, it is easy to parallelize other callers by being lazy about calling `Feature.result()`.

Reply:
@Witiko If this only parallelizes `_collect_dir`, then let's only parallelize it and keep it in the same place. Otherwise we are introducing non-obvious mechanisms for the sake of generalization, that we might never come to using ever again :slightly_smiling_face:

Reply:
@Witiko it won't work generally, because we will need situations where md5 is calculated, but we don't need a result just yet, so we can pass a future around for some time before resolving it. This is not how it is in dvc codebase though, you compare a checksum to something or save it immediately.

Reply:
@efiop Ok, let's start small. We can always parallelize in more places when we need to.